Display correct post updated notices based on post type

earnjam opened the pull request:
Fixes #3315 

In this PR I've added some additional default labels for registered post types following cores existing model of defaulting to `page` for hierarchical and `post` for non-hierarchical:

**Hierarchical** | **Non-Hierarchical**
------------ | -------------
'Page reverted to draft.' | 'Post reverted to draft.'
'Page published.' | 'Post published.'
'Page published privately.' | 'Post published privately.'
'Page scheduled.' | 'Post scheduled.'
'Page updated.' | 'Post updated.'
'View Page.' | 'View post.'


@swissspidy also [suggested to add](https://github.com/WordPress/gutenberg/issues/3315#issuecomment-408424261):

- 'Post submitted for review.'
- 'Post restored to revision from %s.'

However, we don't currently have any type of notices for either of these actions at the moment, so I limited the scope of this PR just to the current notice messages that specified an incorrect post type. 

I wasn't sure if changing `requestPostUpdateSuccess` to `async` was going to have any negative side effects here, but it's how I was able to get this working. Let me know if there are going to be downsides to that I'm unaware of. 

We'll also need to open a Trac ticket to migrate these defaults into `get_post_type_labels()`

earnjam committed:
Add additional register_post_type default labels

earnjam committed:
Use new post type labels for post update success notices

danielbachhuber commented on lib/rest-api.php:
```diff
@@ -460,3 +460,33 @@ function gutenberg_silence_rest_errors() {
+			'reverted_to_draft'   => __( 'Page reverted to draft.' ),
+			'scheduled'           => __( 'Page scheduled.' ),
+			'updated'             => __( 'Page updated.' ),
+			'view'                => __( 'View page.' ),
```
For `'view'` specifically, it looks like we already have a `'view_item'` label we should reuse.

For the others, I think the keys need to be a bit more specific. What about `'published_item'`, `'published_item_privately'`, etc.?

danielbachhuber reviewed (commented):
👍 Great start! One note about the labels. I'll defer to someone else about the JavaScript question.

danielbachhuber commented:
> We'll also need to open a Trac ticket to migrate these defaults into `get_post_type_labels()`

Feel free to create one. I don't believe one exists yet.

earnjam commented:
Associated Trac ticket: https://core.trac.wordpress.org/ticket/45101

earnjam committed:
Update label keys

earnjam committed:
Add docbloc info

earnjam committed:
Fix PHPCS errors/warnings

earnjam committed:
Fix undefined index in PHPUnit test

earnjam committed:
Update strings in tests

earnjam committed:
Remove async call on requestPostUpdateSuccess

danielbachhuber commented on lib/rest-api.php:
```diff
@@ -460,3 +460,36 @@ function gutenberg_silence_rest_errors() {
+			'published_item_privately' => __( 'Page published privately.', 'gutenberg' ),
+			'reverted_item_to_draft'   => __( 'Page reverted to draft.', 'gutenberg' ),
+			'scheduled_item'           => __( 'Page scheduled.', 'gutenberg' ),
+			'updated_item'             => __( 'Page updated.', 'gutenberg' ),
```
@swissspidy @ocean90 Do you have any opinions about the naming of these label keys? (e.g. `published_item`)

ocean90 replied:
I'd probably name it `item_published` as the label itself.

swissspidy replied:
Agreed. This would be more in line with existing keys in `get_post_type_labels()` (`view_item`, `search_items`, `use_featured_image`, etc.).

SO I'd use `item_published`, `item_reverted_to_draft`, `item_scheduled`, etc.

earnjam committed:
Remove period from string in test

earnjam committed:
Rename label keys to use passive voice

earnjam commented:
I adjusted the labels based on the feedback. I also removed `async` from `requestPostUpdateSuccess` and instead just passed `postType` to it from `requestPostUpdate`, since it was causing a test failure.

danielbachhuber commented:
@earnjam Want to go ahead and prepare a corresponding core patch?

earnjam commented:
@danielbachhuber Yep, I'll work on that. Still have a test failure I'm trying to sort out. Might need some other eyes on it, since I'm not too familiar with the JS tests here. It seems to be working fine outside the tests. 

Tested by adding a custom post type with the new labels.

danielbachhuber commented:
@earnjam Does the test suite mock need to be updated to include a mocked `postType` object?

earnjam committed:
Add post type mock data to test

earnjam commented:
I need a facepalm reaction on here for when I miss something obvious. 😆 Thanks @danielbachhuber. 

Ok, I think we should be good to go here now once these tests finish running.

kadamwhite commented:
Looks great. Only 11th-hour rock to toss in: we technically use `messages` to refer to this sort of string elsewhere in core, e.g. `post_updated_messages`, rather than `labels`. I'm willing to be overruled but I'd be interested in preserving that distinction here, using `messages` as our noun.

earnjam commented:
@kadamwhite In this case, these are the `labels` that are passed as args to `register_post_type()`. They show on the REST API endpoint as `labels`. Where in the code do you think we should refer to them as messages?

danielbachhuber commented:
In the interest of simplicity, I'd like to keep these in `labels`, for which there's already prior art.

danielbachhuber committed:
Merge branch 'master' into 3315-post-type-labels

danielbachhuber committed:
Merge branch 'master' into 3315-post-type-labels

mtias commented:
Thank you!
